lib/ddtrace/contrib/auto_instrument.rb

```diff
@@ -25,7 +25,7 @@ def add_auto_instrument
       def self.patch_all
         integrations = []
 
-        Datadog.registry.each do |integration|
+        Contrib::REGISTRY.each do |integration|
           # some instrumentations are automatically enabled when the `rails` instrumentation is enabled,
           # patching them on their own automatically outside of the rails integration context would
           # cause undesirable service naming, so we exclude them based their auto_instrument? setting.
```

Comment:
I think those are two different things:

I moved it to reference `Contrib::REGISTRY` because it's simple and safer, but `c.use integration_name` bellow cannot be done (today) without rebuilding the tracer.

Because auto instrumentation can be triggered at any point the application lifetime, we have to reconfigure components here, as we don't know if this is early or late in the application lifetime. We know for sure this is not doing during tracer initialization, as a recursive `Datadog.configure` call would fail.